Cpp_test/lib/ipc-pipe-send.cpp

```diff
@@ -18,8 +18,20 @@ void IPCPipeSend::init()
 		throw std::runtime_error(std::string("ERROR: create pipe with send name ") + this->info.method_name);
 	}
 
-	// open the pipe
-	this->pd = open(this->info.method_name, O_WRONLY);
+	int time_wait{0};
+	for (time_wait = 0; time_wait < 10; time_wait++) {
+        std::cout << time_wait << std::endl;
+
+        if ((this->pd = open(this->info.method_name, O_NONBLOCK | O_WRONLY)) != -1) {
```

Comment:
As you are using the tag O_NONBLOCK, the sender will not block when writting in the pipe when it is full, the write() function will return -1 and the error can be seen with `errno`. 

You can have more information in EAGAIN error in: https://man7.org/linux/man-pages/man2/write.2.html#ERRORS

The same issue occurs with the receiver when read() should blocked, but with O_NONBLOCK enabled it will just return -1. https://man7.org/linux/man-pages/man2/read.2.html#ERRORS